org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/javaeditor/SemanticHighlightingManager.java

```diff
@@ -432,42 +442,56 @@ private boolean isEnabled() {
 	private void initializeHighlightings() {
 		fSemanticHighlightings= SemanticHighlightings.getSemanticHighlightings();
 		fHighlightings= new Highlighting[fSemanticHighlightings.length];
-
 		for (int i= 0, n= fSemanticHighlightings.length; i < n; i++) {
 			SemanticHighlighting semanticHighlighting= fSemanticHighlightings[i];
-			String colorKey= SemanticHighlightings.getColorPreferenceKey(semanticHighlighting);
-			addColor(colorKey);
 
+			String colorKey= SemanticHighlightings.getColorPreferenceKey(semanticHighlighting);
 			String boldKey= SemanticHighlightings.getBoldPreferenceKey(semanticHighlighting);
-			int style= fPreferenceStore.getBoolean(boldKey) ? SWT.BOLD : SWT.NORMAL;
-
 			String italicKey= SemanticHighlightings.getItalicPreferenceKey(semanticHighlighting);
-			if (fPreferenceStore.getBoolean(italicKey))
-				style |= SWT.ITALIC;
-
 			String strikethroughKey= SemanticHighlightings.getStrikethroughPreferenceKey(semanticHighlighting);
-			if (fPreferenceStore.getBoolean(strikethroughKey))
-				style |= TextAttribute.STRIKETHROUGH;
-
 			String underlineKey= SemanticHighlightings.getUnderlinePreferenceKey(semanticHighlighting);
-			if (fPreferenceStore.getBoolean(underlineKey))
-				style |= TextAttribute.UNDERLINE;
-
 			boolean isEnabled= fPreferenceStore.getBoolean(SemanticHighlightings.getEnabledPreferenceKey(semanticHighlighting));
 
-			fHighlightings[i]= new Highlighting(new TextAttribute(fColorManager.getColor(PreferenceConverter.getColor(fPreferenceStore, colorKey)), null, style), isEnabled);
+			fHighlightings[i]= createHighlightingFromPereferences(semanticHighlighting.getPreferenceKey(), isEnabled, colorKey, boldKey, italicKey, strikethroughKey, underlineKey);
+		}
+
+		SyntaxColorHighlighting[] syntaxHighlightings= SyntaxColorHighlighting.getSyntaxColorHighlightings();
```

Comment:
please autoformat the added lines so that it has uniform space count before assignment operator.